packages/components/data-table-manager/src/data-table-manager.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  useMemo,
+  cloneElement,
+  type ReactElement,
+  type ReactNode,
+  type MouseEventHandler,
+} from 'react';
+import Spacings from '@commercetools-uikit/spacings';
+import DataTableSettings, {
+  type TDataTableSettingsProps,
+} from './data-table-settings';
+
+type TColumProps = {
+  /**
+   * The unique key of the column that is used to identify your data type.
+   * You can use this value to determine which value from a row item should be rendered.
+   * <br>
+   * For example, if the data is a list of users, where each user has a `firstName` property,
+   * the column key should be `firstName`, which renders the correct value by default.
+   * The key can also be some custom or computed value, in which case you need to provide
+   * an explicit mapping of the value by implementing either the `itemRendered` function or
+   * the column-specific `renderItem` function.
+   */
+  key: string;
+
+  /**
+   * The label of the column that will be shown on the column header.
+   */
+  label: ReactNode;
+
+  /**
+   * Sets a width for this column. Accepts the same values as the ones specified for
+   * individual [grid-template-columns](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns).
+   * <br>
+   * For example, using `minmax` pairs (e.g. `minmax(200px, 400px)`), a combinations of
+   * fraction values (`1fr`/`2fr`/etc), or fixed values such as `200px`.
+   * By default, the column grows according to the content and respecting the total table available width.
+   *
+   * @@defaultValue@@: auto
+   */
+  width?: string;
+
+  /**
+   * Use this to override the table's own `horizontalCellAlignment` prop for this specific column.
+   */
+  align?: 'left' | 'center' | 'right';
+
+  /**
+   * A callback function, called when the header cell is clicked.
+   */
+  onClick?: (event: MouseEventHandler) => void;
+
+  /**
+   * A callback function to render the content of cells under this column, overriding
+   * the default `itemRenderer` prop of the table.
+   */
+  renderItem?: (row: ReactNode, isRowCollapsed: boolean) => ReactNode;
```

Comment:
I found another place where we type this and perhaps Rotimi's approach in [here](https://github.com/commercetools/ui-kit/blob/23fd9e859ac86dbe778100ddb10cb9464de53d0c/packages/components/data-table/src/data-table.tsx#L110) is worth following

Reply:
Thanks Kacper.
I'm not sure about this. Linked code is using and internal type which is a styled component but that type is not exposed.
I can't see how should I apply that pattern over here.
Could you provide an example of how you would change this?

Reply:
Thanks Carlos.
I am not sure about it, but my point is that it seems we are changing the signature here from `(row: object, isRowCollapsed: boolean) => React.Node`.
And there was an [issue raised](https://github.com/commercetools/ui-kit/issues/2023) followed by a [PR](https://github.com/commercetools/ui-kit/pull/2024) in the past regarding using non-flexible type of `row`.
Although I am not sure if that applies in this context.